terracotta/handlers/rgb.py

```diff
@@ -105,3 +106,20 @@ def get_band_future(band_key: str) -> Future:
 
     out = np.ma.stack(out_arrays, axis=-1)
     return image.array_to_png(out)
+
+
+def get_scale(scale: NumberOrString, metadata) -> Number:
+    if isinstance(scale, (int, float)):
+        return scale
+    if isinstance(scale, str):
+        # can be a percentile
+        if scale.startswith("p"):
+            # TODO check if percentile is in range
+            percentile = int(scale[1:]) - 1
+            return metadata["percentiles"][percentile]
+
+        # can be a number
+        return float(scale)
```

Comment:
Can it though? I don't think passing a string should be valid if not using the percentile feature.